### PR TITLE
Update operator-sdk to 1.22.1

### PR DESCRIPTION
**Pull request by bcrochet**
In doing so, this patch also goes back to 1.15.0 (which I think is
where we started), and applies all of the migrations that are specified
at https://sdk.operatorframework.io/docs/upgrading-sdk-version/
It should be noted that in the future, this should be check every
time operator-sdk is updated here.

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -308,14 +312,20 @@ spec:
                 - --secure-listen-address=0.0.0.0:8443
                 - --upstream=http://127.0.0.1:8080/
                 - --logtostderr=true
-                - --v=10
-                image: registry.redhat.io/openshift4/ose-kube-rbac-proxy@sha256:f000432f07cd187469f0310e3ed9dcf9a5db2be14b8bab9c5293dd1ee8518176
+                - --v=0
+                image: gcr.io/kubebuilder/kube-rbac-proxy:v0.11.0
```
Comment (komish) [inline bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:316]: Doesn't this still need to be the `ose-kube-rbac-proxy` right?

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -308,14 +312,20 @@ spec:
...
                   name: https
                   protocol: TCP
-                resources: {}
+                resources:
+                  limits:
+                    cpu: 500m
+                    memory: 128Mi
+                  requests:
+                    cpu: 5m
+                    memory: 64Mi
```
Comment (komish) [inline bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:328]: do you want these limits?

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -325,7 +335,7 @@ spec:
                 env:
                 - name: GIT_REPO_PATH
                   value: /git
-                image: quay.io/opdev/operator-certification-operator:1.0.2
+                image: quay.io/opdev/operator-certification-operator:latest
```
Comment (komish) [inline bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:338]: Is using the `latest` tag intentional?

File: config/default/manager_auth_proxy_patch.yaml [hunk 4]
```diff
@@ -10,16 +10,23 @@ spec:
...
         - containerPort: 8443
           protocol: TCP
           name: https
+        resources:
+          limits:
+            cpu: 500m
+            memory: 128Mi
+          requests:
+            cpu: 5m
+            memory: 64Mi
```
Comment (komish) [inline config/default/manager_auth_proxy_patch.yaml:29]: resources, same note as before

File: config/manager/kustomization.yaml [hunk 5]
```diff
@@ -13,4 +13,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: quay.io/opdev/operator-certification-operator
-  newTag: 1.0.2
+  newTag: latest
```
Comment (komish) [inline config/manager/kustomization.yaml:16]: tag, same note as before

**Review by komish — CHANGES_REQUESTED** [review]
Added some notes of things I think we may not be intending to change. The changes I noted may have been intentional, but I'm just remembering back to the times where I bump opsdk and have to manually go through and fix these things.

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (bcrochet) [reply bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:316]: My stance on this:

This should be considered "upstream". Which means, it should "just work". If a brand new operator were to be scaffolded by operator-sdk, this is what you would get. The change to ose-kube-rbac-proxy would happen at the pinning step.

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (bcrochet) [reply bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:338]: Yes. A developer should not need to pass anything to make (other than maybe IMAGE_BUILDER=podman), in order for there to be no unexpected changes to the codebase, leaving confusion as to what should be committed.

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml [hunk 6]
```diff
@@ -419,4 +429,4 @@ spec:
   minKubeVersion: 1.21.0
   provider:
     name: Red Hat
-  version: 1.0.2
+  version: 0.0.0
```
[the remark this answers is not in the document]
Comment (bcrochet) [reply bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:432]: See above comment.

File: config/default/manager_auth_proxy_patch.yaml [hunk 7]
```diff
@@ -10,16 +10,23 @@ spec:
     spec:
       containers:
       - name: kube-rbac-proxy
-        image: registry.redhat.io/openshift4/ose-kube-rbac-proxy@sha256:f000432f07cd187469f0310e3ed9dcf9a5db2be14b8bab9c5293dd1ee8518176
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.11.0
```
[the remark this answers is not in the document]
Comment (bcrochet) [reply config/default/manager_auth_proxy_patch.yaml:13]: See above.

File: config/manager/kustomization.yaml (hunk 5 above)
Comment (bcrochet) [reply config/manager/kustomization.yaml:16]: See above.

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (komish) [reply bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:316]: 👍

**Review by komish — APPROVED** [review]
/lgtm

File: bundle/metadata/annotations.yaml [hunk 8]
```diff
@@ -6,13 +6,10 @@ annotations:
...
-  operators.operatorframework.io.metrics.builder: operator-sdk-v1.14.0+git
+  operators.operatorframework.io.metrics.builder: operator-sdk-v1.22.0
   operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
   operators.operatorframework.io.metrics.project_layout: go.kubebuilder.io/v3
 
   # Annotations for testing.
   operators.operatorframework.io.test.mediatype.v1: scorecard+v1
   operators.operatorframework.io.test.config.v1: tests/scorecard/
-
-  # OpenShift annotations.
```
Comment (acornett21) [inline bundle/metadata/annotations.yaml:17]: Do we see these as only part of the certification process to, and thus removed?

File: bundle.Dockerfile [hunk 9]
```diff
@@ -7,17 +7,14 @@ LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
...
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.14.0+git
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.22.0
 LABEL operators.operatorframework.io.metrics.mediatype.v1=metrics+v1
 LABEL operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v3
 
 # Labels for testing.
 LABEL operators.operatorframework.io.test.mediatype.v1=scorecard+v1
 LABEL operators.operatorframework.io.test.config.v1=tests/scorecard/
 
-# OpenShift annotations.
```
Comment (acornett21) [inline bundle.Dockerfile:18]: same comment as annotations.yaml

File: bundle/metadata/dependencies.yaml [hunk 10]
```diff
@@ -2,4 +2,4 @@ dependencies:
   - type: olm.package
     value:
       packageName: openshift-pipelines-operator-rh
-      version: "1.5.2"
+      version: "1.7.1"
```
Comment (acornett21) [inline bundle/metadata/dependencies.yaml:5]: We need to test this version with our operator as a bundle/catalog in a net new cluster, to make sure the issues in 1.6.x versions were actually fixed.

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (acornett21) [reply bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:316]: @bcrochet What pinning step are you referring to?

File: bundle/metadata/dependencies.yaml (hunk 10 above)
Comment (bcrochet) [reply bundle/metadata/dependencies.yaml:5]: I ran some local tests on a brand-new SNO cluster. I was able to do a pipeline run (the operator I passed itself failed the validation) the pipeline run completed.

Comment (acornett21) [reply bundle/metadata/dependencies.yaml]: I was able to test this as well, and OCO installed properly, as did `operator-pipelines`. I poked the cluster some, `events` and `logs` from both `OLM` operators and didn't see any errors/warns, so I think they fixed all the `1.6.x` issues for the things we care about and need.

**Review by acornett21 — APPROVED** [review]

**Review by komish — APPROVED** [review]
/lgtm

**Pull request merged**